types/lodash-fp/lodash-fp.utils.ts

```diff
@@ -0,0 +1,22 @@
+// eslint-disable-next-line lodash-fp/use-fp -- we need these types from base lodash lib
```

Comment:
I think it's unclear what the distinction is between this file and lodash-fp-helpers.ts. And more generally, the intended distinction between types and src/libs/type-utils. Currently, types contains only type declarations for 3rd party code, but this change makes that definition fuzzier. I think I'd suggest merging this into src/libs/lodash-fp-helpers.ts. That would avoid adding any imports from types.

Reply:
Also, it would be nice to have a comment or example explaining the benefit of these over the `@types/lodash` types. Maybe something in the `.types.ts` file showing how `_.curry` allows things it shouldn't?

Reply:
both great points. 

Question on the directory question... I would like to keep the check-types validation for the lodash/fp utils.... I assume we are ok with those staying in [root]/tests and just the actual types and utils all living in src/libs/type-utils ?

Reply:
I'm fine with that. In the future, we could possibly revisit how check-dts is run if we want to move the type validation side by side with the modules in src/libs/type-utils.

Reply:
I think a big consideration is that .ts files inside the src folder that have errors (expected for check-dts) are a big annoyance in most IDE's that will highlight the directory tree on up with red squigglies to say HEY THERE ARE ERRORS in this folder somewhere...  That might keep us with the very useful check-dts stuff outside of the main src folder.